### ISSUE TITLE
Images should have alt text or alt="" (as should role="img")

### ISSUE THREAD

**Issue by mgifford** [mastodon/mastodon#20445]
### Steps to reproduce the problem

If you run https://accessibilityinsights.io/ or axe DevTools on https://mastodon.social/notifications you will get a bunch of errors related to images and text alternatives.

### Expected behaviour

In this case I think that the images are purely decorative.

### Actual behaviour

Assitive Technology will think this is an image.

### Detailed description

Accessibility errors were introduced by using ARIA.  Right now you have:
`<i role="img" class="fa fa-user-plus fa-fw"></i>`

But really it should be:
`<i class="fa fa-user-plus fa-fw"></i>`

Same goes for many of these other icons which have the purpose repeated in text right beside them. 

If there was a semantic reason for those icons, then they should be described, but they seem purely redundant. 

### Specifications

https://mastodon.social/notifications

**Comment by mgifford** [comment]
Wanted to draw attention to https://github.com/mastodon/mastodon/pull/1424/files

In this PR there was also a label to provide a textual equivalent for the image. That is fine, if there is a label, but the examples in the /notifications didn't have one.

**Comment by erbridge** [comment]
The icons all share a common `Icon` component type. There are two obvious general approaches here:

1. If no `alt`, `title`, `aria-label`, `aria-labelledby` etc is supplied, don't set `role=img`
2. Default `alt` to the `id`, which is used to construct the Font Awesome class (`fa-{id}`) and usually (always?) describes the image, and add a prop to override to say to make it decorative only

Either approach is going to impact other places `Icon` is used (which is a lot). I prefer option 2.

**Comment by mgifford** [comment]
Can you give me an example of what No. 2 looks like. I can't envision what this would look like in HTML.

If there is going to be alt text or a `<span class="element-invisible">fa-{id}</a>` for an Icon, the description has to be human-readable. It can't be, "fa-person" as that isn't helpful. fa-user-plus is a pretty terrible thing to read to someone and it does not add to the text that is immediately beside it (from a semantic point of view):

<img width="841" alt="Screen Shot of Font Awesome ID" src="https://user-images.githubusercontent.com/116832/201441757-c1df00b7-864e-4bdd-a23f-ce078499331b.png">

**Comment by erbridge** [comment]
So in the case of your screenshot, you'd have: `<i role="img" class="fa fa-user-plus fa-fw" alt="user-plus">`

**Comment by mgifford** [comment]
@erbridge `alt="user-plus"` is bad alt text. This has been tried as a fallback with image names. If you stripped out the - it would be a bit better `alt="user plus"` but that would very much depend on how clearly the Font Awesome icon was defined, and how clearly it aligns with what would give users assistance. 

In this case it is a visual cue which is essentially repeated in text so `<i class="fa fa-user-plus fa-fw">` would be best as you said.

**Comment by erbridge** [comment]
There are 2 cases to think about here, I think:

- Icon has semantic value and should be described
- Icon has no semantic value and should be ignored

If we assume we'll need to take an incremental path to making things better (there are a lot of icons affected by this across the service), while also making the defaults better than they are currently, we need to choose a default. The obvious defaults are:

1. Derive alt text from information we already have about the icon, accepting that it won't always be **good** alt text, but will at least not be undescribed.
2. Hide all icons without explicitly set alt text from accessibility tools, accepting that that might create voids where the icon was the only piece of markup, hiding functionality entirely.

In both cases, it would be better, of course, if we also took more care about determining the semantic value of icons across the service now and in the future, and that could be a follow on action. I definitely don't think what I'm suggesting is the whole answer, but it would be good to have better defaults than the current ones (which result in the current behaviour of the service, and the subject of this issue).

I'm not a user of screen readers, but my understanding suggests that (1) is "safer" than (2). My gut says that it's better to announce a thing poorly or superfluously than to hide something important.

But maybe I'm wrong and it's actually better to leave things as they are as the default? Again, in either case, there's the same follow up action, which is to review every icon and make a decision about what changes to make, so this is just about improving the defaults / fallback.

**Comment by Gargron** [comment]
Fixed in #20600
